src/adapters/parsers.ts

```diff
@@ -2,6 +2,10 @@ import { z } from "zod";
 
 import { StrictSchema } from "src/utils/type-safety";
 import { Env, RouterState } from "src/domain";
+import {
+  GetFiatExchangeRatesError,
+  GetFiatExchangeRatesResponse,
+} from "src/adapters/fiat-exchange-rates";
```

Comment:
I don't understand what you mean by `cross-dependency coupling`. I was using this file to declare all the parsers.

Reply:
Sorry, I just made up this `cross-dependency coupling` term and I expect you to follow...

What I mean is:

We're declaring a couple of parsers in `parsers.ts` that are only relevant to (will only be imported by) `fiat-exchange-rates.ts`, but to create them, we need to import the types `GetFiatExchangeRatesError` and `GetFiatExchangeRatesResponse` from `fiat-exchange-rates.ts` (we need them for the `StrictSchema`).

What I mean by `cross-dependency coupling` is that we are coupling these 2 modules in a circular manner (not hierarchical), since both of them import from the other. This is a [code smell](https://martinfowler.com/bliki/CodeSmell.html) that makes me think that:

> We should consider declaring in `parsers.ts` **only the parsers that are reused by more than one module** of the application and move to the consumer module (in this case `fiat-exchange-rates.ts`) those parsers that only affect that module.

In other words, I'd move the declaration of the parsers `fiatExchangeRatesErrorParser` and ` fiatExchangeRatesParser` to `fiat-exchange-rates.ts`.

Reply:
Ah right, I agree! I was using this as a container for all the parsers by the moment. I think that this is outside of the scope of this PR, but I'll move all of them to its module if you want.

Reply:
Done